### ISSUE TITLE
[Bug]: Some icons are not correctly restored in the correct section of the menu bar layout

### ISSUE THREAD

**Issue by ljuillerat** [jordanbaird/Ice#387]
### Check Existing Issues

- [X] I have checked existing issues, and this issue is not a duplicate

### Description

Each time I restart app like 1Password, the icon is set to the "Always Hidden section". I tried to assign it to the "Visible section", but as soon as the app is restarted, the icon is gone.

### Steps to Reproduce

1. Open Ice settings and set 1Password icon to visible section
2. Close settings
3. Close 1Password from menu bar
4. Quit Ice from menu bar
5. Open 1Password app
6. Open Ice
7. Open Ice settings and show menu bar layout

### Ice Version

0.11.6

### macOS Version

15.0 (24A335)

### Screenshots

After step 1:
![image](https://github.com/user-attachments/assets/1840fce1-5acb-4c75-8775-21ebe03fa7e3)
After step 7:
![image](https://github.com/user-attachments/assets/18bcc1d6-51b1-4c88-bdf3-e17c262101f3)

**Comment by jordanbaird** [comment]
Duplicate of #344
